records/generator.go

```diff
@@ -27,9 +29,17 @@ type rrs map[string][]string
 // RecordGenerator contains DNS records and methods to access and manipulate
 // them. TODO(kozyraki): Refactor when discovery id is available.
 type RecordGenerator struct {
-	As       rrs
-	SRVs     rrs
-	SlaveIPs map[string]string
+	As         rrs
+	SRVs       rrs
+	SlaveIPs   map[string]string
+	httpClient http.Client
+}
+
+// NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
+func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
```

Comment:
hm .. let's re-use it in a separate PR

On Thu, Nov 12, 2015 at 7:37 PM, Sargun Dhillon <notifications@github.com>
wrote:

> In records/generator.go
> <https://github.com/mesosphere/mesos-dns/pull/341#discussion_r44736755>:
>
> > @@ -27,9 +29,17 @@ type rrs map[string][]string
> >  // RecordGenerator contains DNS records and methods to access and manipulate
> >  // them. TODO(kozyraki): Refactor when discovery id is available.
> >  type RecordGenerator struct {
> > -	As       rrs
> > -	SRVs     rrs
> > -	SlaveIPs map[string]string
> > +	As         rrs
> > +	SRVs       rrs
> > +	SlaveIPs   map[string]string
> > +	httpClient http.Client
> > +}
> > +
> > +// NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
> > +func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
>
> We weren't reusing the HTTP client before. Is there any reason that was
> the case? I personally don't know, but, if you're confident it's safe to do
> it, I'll make the change to have it passed in.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/341/files#r44736755>.
>